core/services/relay/evm/binding/named.go

```
@@ -0,0 +1,263 @@
+				}.ReadIdentifier(methodResult.MethodName),
+			}
+
+			brr.SetResult(methodResult.ReturnValue, methodResult.Err)
+			batchGetLatestValuesResults[contractName] = append(batchGetLatestValuesResults[contractName], brr)
+		}
+	}
+
+	return batchGetLatestValuesResults, err
+}
+
+// Unbind unbinds contract addresses to contract bindings and read bindings.
+func (b *NamedBindings) Unbind(ctx context.Context, reg Registrar, bindings []commontypes.BoundContract) error {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	for _, binding := range bindings {
+		contract, exists := b.contractBindings[binding.Name]
+		if !exists {
+			return fmt.Errorf("%w: no contract named %s", commontypes.ErrInvalidConfig, binding.Name)
+		}
+
+		b.contractLookup.unbindAddressForContract(binding.Name, binding.Address)
+
+		if err := errors.Join(
+			contract.Unbind(ctx, reg, common.HexToAddress(binding.Address)),
+			contract.UnbindReaders(ctx, common.HexToAddress(binding.Address)),
+		); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *NamedBindings) RegisterAll(ctx context.Context, reg Registrar) error {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	for _, cb := range b.contractBindings {
+		if err := errors.Join(cb.RegisterReaders(ctx), cb.Register(ctx, reg)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *NamedBindings) UnregisterAll(ctx context.Context, reg Registrar) error {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	for _, cb := range b.contractBindings {
+		if err := errors.Join(cb.UnregisterReaders(ctx), cb.Unregister(ctx, reg)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *NamedBindings) SetCodecAll(codec commontypes.RemoteCodec) {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	for _, cb := range b.contractBindings {
+		cb.SetCodecAll(codec)
+	}
+}
+
+func (b *NamedBindings) WithFilter(name string, filter logpoller.Filter) {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	if contract, ok := b.contractBindings[name]; ok {
```

This should throw an error, if we get to this point something went very wrong because we either have a contract without methods or events or a disappearing map entry